### PR TITLE
Implement CompletionItemLabel to improve code completion UX

### LINKED ISSUE #153: Implement CompletionItemLabel
Right now the it only gives 1 completion detail for the selected item, however we should be able to see details for all the items in the completion list

VScode recently added support for it, we should implement it

https://github.com/microsoft/vscode/issues/39441#issuecomment-869628497

(there must be more to support having detail for all items, but idk what yet)

It makes a huge difference, for ex in CLion:

image

You can instantly see what you want

### PR DESCRIPTION
**Pull request by RUSshy**
I messed up the previous PR, here is a clean one
Fixes #153 

**BEFORE:**

Code_gDQzNqNv7q

**AFTER:**

Code_9KZhCUV46L

### CODE REVIEW

File: source/served/commands/complete.d [hunk 1]
```diff
@@ -563,7 +568,11 @@ CompletionList provideDietSourceComplete(TextDocumentPositionParams params,
 			ret.label = a.text;
 			ret.kind = a.type.mapToCompletionItemKind.opt;
 			if (a.definition.length)
+			{
 				ret.detail = a.definition.opt;
+				if (capabilities.textDocument.completion.completionItem.labelDetailsSupport)
+					ret.labelDetails = CompletionItemLabelDetails(ret.detail);
```
Comment (WebFreak001) [inline source/served/commands/complete.d:574]: diet might need some testing and the change here could either be left out or be implemented properly

File: source/served/commands/complete.d [hunk 2]
```diff
@@ -924,21 +935,131 @@ unittest
...
 		{
 			item.detail = identifier.definition;
-			if (!completeNoDupes)
-				item.sortText = identifier.definition;
+
+			// check if that's actually a proper completion item to process
+			auto definitionSpace = identifier.definition.indexOf(' ');
+			if (definitionSpace != -1)
+			{
+				detailDescription = identifier.definition[0 .. definitionSpace];
```
Comment (WebFreak001) [inline source/served/commands/complete.d:1023]: this should have a comment explaining a few cases what this does / how this looks (as just dropping everything after space could be suboptimal)

File: source/served/commands/complete.d [hunk 3]
```diff
@@ -924,21 +935,131 @@ unittest
...
+				auto retTypeEnd = identifier.definition.lastIndexOf(' ', nameEnd);
+				if (retTypeEnd != -1)
+					detailDescription = identifier.definition[0 .. retTypeEnd].strip;
+				else
+					detailDescription = "auto";
+
+				detailDetail = " " ~ identifier.definition[nameEnd .. $];
+			}
+
+			item.sortText = identifier.definition;
```
Comment (WebFreak001) [inline source/served/commands/complete.d:1061]: unrelated (was test)

**Review by WebFreak001 — COMMENTED** [review]
note to self

**Comment by RUSshy** [comment]
in ignorance we live

**Comment by WebFreak001** [comment]
why did you close this?

**Pull request closed without merging**